tx/textual/valuerenderer/int.go

```diff
@@ -26,5 +28,33 @@ func (vr intValueRenderer) Format(_ context.Context, v protoreflect.Value) ([]Sc
 }
 
 func (vr intValueRenderer) Parse(_ context.Context, screens []Screen) (protoreflect.Value, error) {
-	panic("implement me")
+	if len(screens) != 1 {
+		return protoreflect.Value{}, fmt.Errorf("expected single screen: %v", screens)
+	}
+
+	parsedInt, err := parseInt(screens[0].Text)
+	if err != nil {
+		return protoreflect.Value{}, err
+	}
+
+	return protoreflect.ValueOfString(parsedInt), nil
+}
+
+// parseInt formats an value-rendered string into an integer
+func parseInt(v string) (string, error) {
+	sign := ""
+	if v[0] == '-' {
+		sign = "-"
+		v = v[1:]
+	}
+
+	v = strings.TrimLeft(v, "0")
+	if v == "" {
+		return "0", nil
+	}
```

Comment:
```suggestion
```

We only need to parse value-rendered strings. There's no need to trimLeft in those cases